src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -208,9 +210,15 @@ public class List<T> : IList<T>, System.Collections.IList, IReadOnlyList<T>
         // increased by one. If required, the capacity of the list is doubled
         // before adding the new element.
         //
-        public void Add(T item) {
-            if (_size == _items.Length) EnsureCapacity(_size + 1);
-            _items[_size++] = item;
+        public void Add(T item)
+        {
+            var size = _size;
+            if (size == _items.Length)
+            {
+                EnsureCapacity(size + 1);
```

Comment:
Tried something like @stephentoub's https://github.com/dotnet/coreclr/pull/9323 where the addition was also factored out to try and get `Add` to be inlinable as its not; but the simple function was `[below ALWAYS_INLINE size]` and seemed to ignore the `[MethodImpl(MethodImplAttributes.NoInlining)]` attribute.

Not sure if is expected or a  change? /cc @AndyAyersMS